### PR TITLE
HPCC-27093 Add ESDL based ECL result monitoring support for MySQL

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by afishbeck** [comment]
@richardkchapman note that this is a change to a "esdl" command line command only used by scapps for creating alert monitors.

The new file "esdl2monitor_cassandra.xslt" file is really just a copy of the old xslt that did not change at all.

The generated code is being tested by scapps next week, so we shouldn't actually merge until their test is complete.

File: esp/xslt/esdl2monitor.xslt [hunk 1]
```diff
@@ -45,8 +44,9 @@
 EXPORT <xsl:value-of select="$definitionName"/>() := MACRO<xsl:text>
 </xsl:text>
 </xsl:if>
+
 <xsl:if test="$diffmode='Monitor'">
-IMPORT cassandra;
+IMPORT mysql;
```
Comment (afishbeck) [inline esp/xslt/esdl2monitor.xslt:49]: Note that Cassandra support has been moved to a different file: `esdl2monitor_cassandra.xslt`

File: esp/xslt/esdl2monitor_cassandra.xslt [hunk 2]
```diff
@@ -0,0 +1,1122 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (afishbeck) [inline esp/xslt/esdl2monitor_cassandra.xslt:1]: Note that this is just a copy of the old esdl2monitor.xslt.  Nothing has changed except the copyright date.

**Comment by afishbeck** [comment]
The ECL and ESP (WsNotify) developers have given a thumbs up to these changes.  They have tested the ECL code generated by this tool for their project.

**Review by asselitx — CHANGES_REQUESTED** [review]
@afishbeck Perhaps just two small changes needed that I've commented on.

File: tools/esdlcmd/esdlcmd_monitor.cpp [hunk 3]
```diff
@@ -1198,6 +1205,7 @@ class EsdlMonitorCmd : public EsdlConvertCmd
         puts("  --hide-get-data-from  Hide ESDL elements with @get_data_from option" );
 
         puts(ESDLOPT_INCLUDE_PATH_USAGE);
+        puts("   --use-cassandra                        Gerate monitoring code that stores it's state in Cassandra (default is MySQL)");
```
[the remark this answers is not in the document]
Comment (afishbeck) [reply tools/esdlcmd/esdlcmd_monitor.cpp]: I've fixed the spelling.

File: esp/xslt/esdl2monitor.xslt [hunk 4]
```diff
@@ -26,7 +26,6 @@
     <xsl:param name="diffmode" select="'Monitor'"/>
```
[the remark this answers is not in the document]
Comment (afishbeck) [reply esp/xslt/esdl2monitor.xslt:26]: We don't typically update the copyright when we make code changes to a file.

**Comment by afishbeck** [comment]
@asselitx updated the spelling.

**Review by asselitx — APPROVED** [review]
@afishbeck  looks good.

**Comment by afishbeck** [comment]
@richardkchapman the PR is ready to merge.  It only affects a command line tools used to generate ECL used for alerting/monitoring projects.

**Pull request merged**